lib/universal/mapper.js

```diff
@@ -611,9 +615,24 @@ function createCommonGAForm(facade, settings) {
   if (app.appInstallerId) form.aiid = app.appInstallerId;
 
   if (settings.sendUserId && facade.userId()) form.uid = facade.userId();
-  if (facade.userAgent()) form.ua = facade.userAgent();
   if (facade.ip()) form.uip = facade.ip();
 
+  /**
+ * GA parses userAgent to compute device type.
+ * Generates a user agent if one is not provided for all calls only while context is present.
+ *
+ * Expected format:
+ * Mozilla/[version] ([system and browser information]) [platform] ([platform details]) [extensions]
+
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ua
+ */
+
+  if (facade.userAgent()) {
+    form.ua = facade.userAgent();
+  } else if(!facade.userAgent() && osName && osVersion && deviceModel && deviceManufacturer && locale) {
```

Comment:
If the `userAgent` doesn't return a value, but I guess that is accounted for in the first statement. Should I remove?